daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/GrpcLedgerClient.scala

```diff
@@ -382,7 +382,7 @@ class GrpcLedgerClient(val grpcClient: LedgerClient, val applicationId: Applicat
       grpcClient.userManagementClient
         .listUsers(pageToken = pageToken, pageSize = pageSize)
         .flatMap { case (users, nextPageToken) =>
-          if (users.size < pageSize) Future.successful(users.toList)
+          if (users.size < pageSize || nextPageToken == "") Future.successful(users.toList)
```

Comment:
```suggestion
          if (nextPageToken == "") Future.successful(users.toList)
```
Seems redundant unless the ledger is buggy and we cannot sensible guard against broken ledgers.

Reply:
I explained on the other comment: https://github.com/digital-asset/daml/pull/12710#discussion_r799538299

> We shouldn't have nextPageToken == "" be the only test, or else every call to listAllUsers will make a final pointless roundtrip over the Grpc interface. But we can add it as an extra check for safety's sake.

Reply:
Ah I see, that seems like something the ledger should maybe fix but agreed on keeping it there. Would be great to preserve your GH comment in a code comment.

Reply:
I'm not entirely sure that's the case though, it sounds to me that it should be the other way around. If you don't have any further result you should receive an empty string as the next page token and you avoid starting another call. If you rely on receiving one final blank page, you are actually issuing an extra call. Or am I missing something?

Reply:
The case Nick is trying to get at is this one:
1. We request for a page size of 100
2. We get one page with 50 users and a next page token.
3. The next page is empty and has no next token.

There is really no reason to make request 3, the ledger should just include no next token on page 2 but apparently that’s not what’s implemented.

Reply:
I don't think you would get a next page token if there are no further results.

https://github.com/digital-asset/daml/blob/6bdb9011270f94d6f8ffc8914ef9dc2fc2f0e059/ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto#L137-L144

Am I misreading?

Reply:
I wrote a code comment to explain.

Reply:
It describes what I witnessed experimentally.

Reply:
Thanks. I don't think that that should be the upstream behavior but thanks for clarifying.